fix(fullname): added extractFullName() template

jameszheng405 opened the pull request:
fix #2539

Fixes #2539.

**Changes proposed in this pull request:**

- added extractFullName() template

_Note: pull requests without proper descriptions may simply be closed without further discussion. We appreciate your contributions, but need to know what you are offering in clearly described format. Provide tests for all code that you add/modify. If you add/modify any components update the storybook. Thanks! (you can delete this text)_

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/HospitalRun/hospitalrun-frontend?pullRequest=2576) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/HospitalRun/hospitalrun-frontend?pullRequest=2576) before we can accept your contribution.<br/><hr/>**James Zheng** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/HospitalRun/hospitalrun-frontend?pullRequest=2576) it.</sub>

morrme commented:
Hi @jameszheng405 ! Thanks for this contribution. 

Have you previously signed our CLA? https://cla-assistant.io/HospitalRun/hospitalrun-frontend?pullRequest=2576

Also, it looks like your commits were not signed by an email tied to your Github account. 
See more information here: https://docs.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

If you need any guidance with sorting these issues out, let us know!

jameszheng405 committed:
fix(extractfullname.ts): scaffolds extractFullName()

Previously, the extractUserName() function was returning a mocked "some user." We scaffolded out
this function for when we complete the user model.

fix #2539

fixed extractFullName import

remove reportedBy

make app test less spurious

DrewGregory commented on src/__tests__/App.test.tsx:
```diff
@@ -32,9 +32,9 @@ it('renders without crashing', async () => {
-  expect(
-    await screen.findByRole('heading', { name: /dashboard\.label/i }, { timeout: 8000 }),
-  ).toBeInTheDocument()
+  await waitFor(() => {
```
This test was failing, so we changed it to match RTL's default timeout behavior, and it now passes.

jameszheng405 commented:
@morrme Can you review this PR again, I believe the tests are now passing
